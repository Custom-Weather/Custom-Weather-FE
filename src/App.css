.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

.background-image{
  height: 100%;
  background-size: cover;
  position: absolute;
  z-index: -1;
  display: flow-root;
}


.weather-components{
  /* background-color: white; */
  padding: 1%;
  display: inline-table;
  height: 100%;
  z-index: 1;
  top: 1%;
}


.page{
  text-align: center;
}

.location-inline-form{
  position: fixed;
  top: 3%;
  left: 2%;
  background-color: white;
  padding: 4px;
}

.form-only {
  text-align: center;
}
