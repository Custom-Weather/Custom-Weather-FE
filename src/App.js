import React, { Component } from 'react';
import Navbar from './Navbar';
import LocationForm from './LocationForm';
import Weather from './Weather';
import BackgroundImage from './BackgroundImage'
import { BrowserRouter, Switch, Route } from 'react-router-dom';
import './App.css';

class App extends Component {
  constructor() {
    super();
    this.state = {
      weather: {
        'current': 'Just Test Data',
          'sunrise': 1586487424,
          'sunset': 1586487424,
          'high': 60,
          'low': 43,
          'desc': 'Thunderstorm',
          "hourly":
          [{"dt": 1588935600,
           "temp": 16.75,
           "weather": [
                         {
                           "main": "Rain",
                           "description": "moderate rain",
                           "icon": "10n"
                         }
                       ]
          },
          {"dt": 1588939200,
           "temp": 16.75,
           "weather": [
                         {
                           "main": "Rain",
                           "description": "moderate rain",
                           "icon": "10n"
                         }
                      ]
                    }
                  ]
      },
      location: ""
        }
    }

  updateWeather = (latlong, location) => {
    console.log(location)
    fetch(`https://cors-anywhere.herokuapp.com/http://weatherbeefy.herokuapp.com/weather/api/v1/${ latlong.lat }&${ latlong.lng }`)
      .then(response => {return response.json()})
      .then((data) => {
        console.log(data)
        this.setState((state) => ({weather: data, location: location}))
      })
  }

  render() {
      if(this.state.weather.current == 'Just Test Data') {return (
        <div className='form-only'>
        <BrowserRouter>
        <Switch>
        <Route path="/" exact component={()=> <LocationForm updateWeather={this.updateWeather}/>}  />
        </Switch>
        </BrowserRouter>
        </div>
        );}
        else {return (
          <main className='App'>
          <BrowserRouter>
          <Switch>
          <Route path="/" exact component={()=> <LocationForm updateWeather={this.updateWeather}/>}  />
          <Route path="/dashboard" render={props =>
            <div className='page'>
              <div className='navbar'>< Navbar /></div>
              <div className='background-image'>
                <BackgroundImage desc={this.state.weather.desc} />
              </div>
              <div className='weather-components'>
                <Weather weather={this.state.weather}/>
              </div>
            </div>} />
            </Switch>
          </BrowserRouter>
          </main>
    );}
  }

}

export default App;
