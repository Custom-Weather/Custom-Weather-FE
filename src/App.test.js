import React from 'react';
import { render, cleanup } from '@testing-library/react';
import App from './App';

afterEach(cleanup)

 it('should take a snapshot', () => {
    const { asFragment } = render(<App />)

    expect(asFragment(<App />)).toMatchSnapshot()
  })

  it('renders welcome message', () => {
    const { getByText } = render(<App />);
    expect(getByText('Learn React')).toBeInTheDocument();
  });

// test('renders learn react link', () => {
//   const { getByText } = render(<App />);
//   const linkElement = getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });
