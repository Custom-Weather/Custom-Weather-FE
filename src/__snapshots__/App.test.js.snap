// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="form-only"
  >
    <div>
      <h1>
        Custom Weather
      </h1>
      <div>
        <form>
          <label>
            Location:
            <input
              data-testid="location-input"
              placeholder="Type in city and state"
              type="text"
            />
          </label>
          <button
            data-testid="input-submit"
            type="submit"
          >
            Submit
          </button>
          <button
            data-testid="current-submit"
            type="submit"
          >
            Use Current Location
          </button>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
